src/app/plugins/ta.plugin.ts

```diff
@@ -0,0 +1,167 @@
+import { Plugin } from '../../common/plugin';
+import { IContainer, IMessage, ChannelType, Maybe } from '../../common/types';
+import { GuildChannel, GuildMember, TextChannel, Role } from 'discord.js';
+
+export class TaPlugin extends Plugin {
+  public name: string = 'TA Plugin';
+  public description: string = 'Allows TAs to register for classes.';
+  public usage: string = 'ta <register/remove> | ta ask <question>';
+  public pluginAlias = [];
+  public permission: ChannelType = ChannelType.Private;
+
+  private _DISCRIMINATOR_LENGTH: number = '#0000'.length;
+  private _TA_ROLE = 'Teaching Assistant';
+  private _roleCache: Record<string, Maybe<Role>> = {
+    [this._TA_ROLE]: undefined,
+  };
+
+  constructor(public container: IContainer) {
+    super();
+  }
+
+  public validate(message: IMessage, args: string[]) {
+    return !!args.length;
+  }
+
+  public async execute(message: IMessage, args?: string[]) {
+    if (!args?.length) {
```

Comment:
we already know args is valid because of line 23

Reply:
it thinks args is `string[] | undefined` inside of execute without that

Reply:
`if(!args){`?